helper/testlog/testlog.go

```diff
@@ -115,3 +119,29 @@ func (w *prefixStderr) Write(p []byte) (int, error) {
 
 	return os.Stderr.Write(buf)
 }
+
+// Option allows modifications to hclog.LoggerOptions when creating
+// a new hclog.Logger.
+type Option func(*hclog.LoggerOptions)
+
+// WithOutput will add the provided writer to the loggers outputs.
+func WithOutput(w io.Writer) Option {
+	return func(opts *hclog.LoggerOptions) {
+		if opts.Output != nil {
+			opts.Output = io.MultiWriter(opts.Output, w)
+		} else {
+			opts.Output = w
+		}
+	}
+}
+
+// WithLogging sets the NOMAD_TEST_LOG_LEVEL to the provided level.
+// After executing the provided function, NOMAD_TEST_LOG_LEVEL is
+// set back to its original value.
+func WithLogging(level string, fn func()) {
+	lvl := os.Getenv("NOMAD_TEST_LOG_LEVEL")
+	os.Setenv("NOMAD_TEST_LOG_LEVEL", level)
+	defer os.Setenv("NOMAD_TEST_LOG_LEVEL", lvl)
```

Comment:
this can have side effects with tests run in parallel.  if you pass a `testing.T` in here, it has a helper to prevent this happening by accident:

https://pkg.go.dev/testing#T.Setenv

(not a blocker, just a suggestion!)

Reply:
you also could avoid passing in the `func()` this way, since `t.Setenv` uses [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) to restore the original env -- that runs at the end of the test rather than a `defer` that must run at the end of this func.